### PR TITLE
[FIX] review OpenSwath interfaces

### PR DESCRIPTION
**Pull request by hroest**
review the OpenSwath interfaces, create some improvements (e.g. const correctness)

### CODE REVIEW

File: src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp [hunk 1]
```diff
@@ -840,7 +840,7 @@ namespace OpenMS
       if (stop_report_after_feature_ >= 0 && i >= (Size)stop_report_after_feature_) {break;}
       output.push_back(feature_list[i]);
     }
-    transition_group = transition_group_detection;
+    transition_group = transition_group_detection; // TODO why is this here?
```
Comment (hroest) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp]: @grosenberger do you know why we are doing this? I dont quite understand the point of overwriting the `transition_group` here?

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMAssay.h [hunk 2]
```diff
@@ -325,12 +338,12 @@ namespace OpenMS
       TODO: why make copy of IonMapT ?
```
Comment (hroest) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMAssay.h]: @grosenberger can you comment on whether we could change pass `DecoyIonMap` and `TargetIonMap` by reference?

Comment (grosenberger) [reply src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMAssay.h]: Yes, we should change that, thanks for spotting this. They are only used as reference and should not be modified.

File: src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp (hunk 1 above)
Comment (hroest) [reply src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp]: ok great, I have changed it and now `transition_group` is `const`

Comment (hroest) [reply src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp]: nope, seems that its important to copy back the information...

**Comment by grosenberger** [comment]
Great, thanks for doing this! 👍

**Comment by hroest** [comment]
I think this can be merged, failing tests are unrelated

**Pull request merged**